lib/util.js

```diff
@@ -68,71 +68,59 @@ exports.format = function(f) {
     return objects.join(' ');
   }
 
-  var argLen = arguments.length;
-
-  if (argLen === 1) return f;
+  if (arguments.length === 1) return f;
 
   var str = '';
   var a = 1;
   var lastPos = 0;
   for (var i = 0; i < f.length;) {
-    if (f.charCodeAt(i) === 37/*'%'*/ && i + 1 < f.length) {
+    if (f.charCodeAt(i) === 37/*'%'*/ && f.length > i + 1) {
+      if (f.charCodeAt(i + 1) !== 37 && a >= arguments.length) {
+        ++i;
+        continue;
+      }
       switch (f.charCodeAt(i + 1)) {
         case 100: // 'd'
-          if (a >= argLen)
-            break;
-          if (lastPos < i)
```

Comment:
Sorry, I didn't understand this point. You meain only remain `lastPos < i` instead of `i > lastPos` ? Thank you!